virttest/qemu_vm.py

```diff
@@ -1293,6 +1293,33 @@ def _get_pci_bus(devices, params, dtype=None, pcie=False):
                     pass
             return {'aobject': params.get('pci_bus', 'pci.0')}
 
+        def add_pci_controllers(devices, params):
+            """
+            Insert pci controllers into qcontainer by order.
+
+            :param devices: qcontainer object
+            :param params: test params
+            """
+            def sort_key(dev):
+                """
+                Function used to sort pcic list
+                """
+                order_pcics = ['ioh3420', 'x3130-upstream',
+                               'x3130', 'i82801b11-bridge',
+                               'pci-bridge']
+                try:
+                    return order_pcics.index(dev.get_param('driver'))
+                except ValueError:
+                    return -1
+
+            pcics = []
+            for pcic in params.objects("pci_controllers"):
+                dev = devices.pcic_by_params(pcic, params.object_params(pcic))
+                slot = len([_ for _ in pcics if _.get_param('driver') == 'ioh3420'])
+                if dev.get_param('driver') == 'ioh3420':
+                    dev.set_param('slot', hex(slot))
+                pcics.append(dev)
```

Comment:
'slot' get value by go through the list 'pcics', but the list 'pcics' is append after that, if first loop 'pcic' has driver 'ioh3420', currently list 'pcics' is empty, I am not sure, looks slot still not assigned value in this case?